swarm/src/connection/pool.rs

```diff
@@ -54,10 +55,44 @@ use void::Void;
 mod concurrent_dial;
 mod task;
 
+enum ExecSwitch<TExecutor>
+where
+    TExecutor: Executor,
+{
+    Executor(TExecutor),
```

Comment:
I would be in favor of using a trait object here. That will get rid of the type parameter which should make the signature changes less invasive!